### PR TITLE
SPORTSHUB-174, SPORTSHUB-183 - Fixing issues with book now on mobile and 0 vacancy

### PR DESCRIPTION
**Pull request by ReggieStar11**
Fixed the mobile view of MobileEventPayment.tsx, added the option to book now, so it works for mobile view, als
o fixed the issue with the 'book now' button appearing when the vacancy of an event is zero, so it just shows the message no tickets avaliable"

## Why was your change needed?

Please detail here why you did the change you did.

## What was Changed?

Please detail here what were the main changes you made.

## Any extra notes for reviewers to know?

Write any extra notes you may have for the PR.
This may include specific code paths that need extra scrutiny, or how to review a specific portion of code.

### CODE REVIEW

File: frontend/components/events/EventPayment.tsx [hunk 1]
```diff
@@ -96,48 +93,50 @@ export default function EventPayment(props: EventPaymentProps) {
...
+                      value={`${attendeeCount}`}
+                      onChange={handleAttendeeCount}
+                      labelProps={{
+                        className: "text-black before:border-black after:border-black",
+                      }}
+                      menuProps={{
+                        className: "text-black",
+                      }}
+                    >
+                      {Array(Math.min(props.vacancy, 7))
```
Comment (owya490) [inline frontend/components/events/EventPayment.tsx:111]: /nit - can we extract this 7 to a constant

**Review by owya490 — APPROVED** [review]
looks good thanks

**Review by owya490 — APPROVED** [review]
lgtm

**Review by 41dan — APPROVED** [review]
tested functionality and lgtm

**Pull request merged**